packages/next-codemod/bin/upgrade.ts

```diff
@@ -160,6 +230,23 @@ function getInstalledNextVersion(): string {
   }
 }
 
+function getInstalledReactVersion(): string {
+  try {
+    return require(
+      require.resolve('react/package.json', {
+        paths: [process.cwd()],
+      })
+    ).version
+  } catch (error) {
+    throw new Error(
+      `Failed to get the installed React version at "${process.cwd()}".\nIf you're using a monorepo, please run this command from the Next.js app directory.`,
```

Comment:
Failed to detect the installed React version in "${process.cwd()}".\nIf you're working in a monorepo, please run this command from the Next.js app directory.